### ISSUE TITLE
ReLU operator for classifier in training from scratch

### ISSUE THREAD

**Issue by anhtienng** [PKU-ML/non_neg#2]
Dear authors,

Thank you for your amazing work. I have a question regarding the training from scratch with NCE.

Do you apply ReLU for the bias of the linear classifier? (according to the manuscript, you should apply the ReLU for embeddings before the classification head and the weight of the head itself).

Thank you.

**Comment by yifeiwang77** [comment]
hi thanks for your interest! there is no bias term in this case

**Comment by anhtienng** [comment]
Thanks for answering. BTW I can't find the learning rate, optimizer,... setting for the training from scratch experiments with NCE. Can you please provide them? Thank you.

**Comment by anhtienng** [comment]
I am trying to use NCE for training from scratch with ResNet50 on ImageNet-1K (without a MLP projector). But it seems like the CE is still better than NCE.
I tried with different seeds for initialization. The ACC_val is as below:
<img width="1901" height="1165" alt="Image" src="https://github.com/user-attachments/assets/73a79d75-067b-4876-9f27-7a443f8972fa" />
This is my simple code for NCE:
```
output_before_head = torch.relu(output_before_head) # ReLU for input of a linear classifier
output_after_head = output_before_head @ torch.relu(self.classifier.weight).t() # ReLU for the weights of a linear classifier
```
Are there any hyper-parameters that I need to carefully tune for NCE?
I am using `bs=1024, lr=0.4, epochs=100, optimizer=SGD, weight-decay=0.00002, scheduler=cosine, lr-warmup-epochs=5`.

SImilar to ResNet18 on ImageNet-1K.
<img width="1955" height="1160" alt="Image" src="https://github.com/user-attachments/assets/4a88486e-7062-4dee-90a2-4b5317d3193e" />
